drivers/display/display_st7789v.c

```diff
@@ -79,57 +72,59 @@ static void st7789v_set_lcd_margins(struct st7789v_data *data,
 	data->y_offset = y_offset;
 }
 
-static void st7789v_set_cmd(struct st7789v_data *data, int is_cmd)
+static void st7789v_set_cmd(struct st7789v_config *config, int is_cmd)
 {
-	gpio_pin_set(data->cmd_data_gpio, ST7789V_CMD_DATA_PIN, is_cmd);
+	gpio_pin_set_dt(&config->cmd_data_gpio, is_cmd);
 }
 
-static void st7789v_transmit(struct st7789v_data *data, uint8_t cmd,
-		uint8_t *tx_data, size_t tx_count)
+static void st7789v_transmit(struct st7789v_data *data, struct st7789v_config *config, uint8_t cmd,
```

Comment:
No, please use `const struct device *dev`
```
static void st7789v_transmit(const struct device *dev, uint8_t cmd,
                             uint8_t *tx_data, size_t tx_count)
```

Same for st7789v_exit_sleep, t7789v_set_mem_area, ...

Reply:
I thought this version is more type-safe. I will change to `device *`.